src/knowledge_sources/NumberOfSourcesKS.m

```diff
@@ -0,0 +1,83 @@
+classdef NumberOfSourcesKS < AuditoryFrontEndDepKS
+    
+    properties (SetAccess = private)
+        modelname;
+        model;                 % classifier model
+        featureCreator;
+        blockCreator;
+    end
+
+    methods
+        function obj = NumberOfSourcesKS( modelName, modelDir, ppRemoveDc )
+            modelFileName = [modelDir filesep modelName];
+            v = load( [modelFileName '.model.mat'] );
+            if ~isa( v.featureCreator, 'FeatureCreators.Base' )
+                error( 'Loaded model''s featureCreator must implement FeatureCreators.Base.' );
+            end
+            afeRequests = v.featureCreator.getAFErequests();
+%             if nargin > 2 && pp_bNormalizeRMS
+%                 for ii = 1 : numel( afeRequests )
+%                     afeRequests{ii}.params.replaceParameters( ...
+%                                    genParStruct( 'pp_bNormalizeRMS', pp_bNormalizeRMS ) );
+%                 end
+%             end
+            if nargin > 2 && ppRemoveDc
+                for ii = 1 : numel( afeRequests )
+                    afeRequests{ii}.params.replaceParameters( ...
+                                   genParStruct( 'pp_bRemoveDC', ppRemoveDc ) );
+                end
+            end
+            obj = obj@AuditoryFrontEndDepKS( afeRequests );
+            obj.featureCreator = v.featureCreator;
+            if isfield(v, 'blockCreator')
+                if ~isa( v.blockCreator, 'BlockCreators.Base' )
+                    error( 'Loaded model''s block creator must implement BeatureCreators.Base.' );
+                end
+            elseif isfield(v, 'blockSize_s')
+                v.blockCreator = BlockCreators.StandardBlockCreator( v.blockSize_s, 0.5/3 );
+            else
+                % for models missing a block creator instance
+                v.blockCreator = BlockCreators.StandardBlockCreator( 0.5, 0.5/3 );
+            end
+            obj.blockCreator = v.blockCreator;

```

Comment:
This would break the `IdentityLocationKS` which also inherits from `IdentityKS` but whose model, `CaffeModel` is no longer derived from `Models.Base`. The inheritance allows easy check and instantiation of the feature creator and the default behavior for pp_bRemoveDC.
It could work if we have the IdentityKS constructor call methods that the derived classes can override.